network/dns_packet.ksy

```diff
@@ -60,19 +60,24 @@ types:
       - id: rdlength
         doc: "Length in octets of the following payload"
         type: u2
-      - id: ptrdname
-        type: domain_name
-        if: "type == type_type::ptr"
-      - id: address
-        type: address
-        if: "type == type_type::a"
+      - id: payload
+        size: rdlength
+        type:
+          switch-on: type
+          cases:
+            'type_type::ptr': domain_name
+            'type_type::a': address
+            'type_type::cname': domain_name
+            'type_type::srv': service
+            'type_type::txt': txt_body
+
   domain_name:
     seq:
       - id: name
         type: label
         repeat: until
         doc: "Repeat until the length is 0 or it is a pointer (bit-hack to get around lack of OR operator)"
-        repeat-until: "_.length == 0 or _.length == 0b1100_0000"
+        repeat-until: "_.length == 0 or _.length >= 192"
```

Comment:
Shouldn't it be `_.length == 0 or _.is_pointer` then, as we already have this `is_pointer` set up?